### PR TITLE
(PC-5001) update to react-query3.1.1

### PR DESCRIPTION
**Pull request by Giregm**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-5001

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn version:testing` (this will create a commit with a tag)
- then run `git push --follow-tags`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     "react-native-splash-screen": "^3.2.0",
     "react-native-svg": "^12.1.0",
     "react-native-webview": "^10.10.0",
-    "react-query": "^2.23.1",
+    "react-query": "3.1.1",
```
Comment (antoinewg) [inline package.json:71]: J'ai l'impression qu'on peut supprimer "@types/react-query", vu que les types sont dans react-query.

File: package.json [hunk 2]
```diff
@@ -68,7 +68,7 @@
...
     "recompose": "^0.30.0",
     "styled-components": "^5.2.0",
```
Comment (antoinewg) [inline package.json:73]: dans le package.json de react-query-native-devtools:

image

⚠️ Si on passe à 3.1.1, on dit adieu au devtools. Je suis pas contre mais à voir si qqn l'utilise.

File: src/features/home/pages/Home.test.tsx [hunk 3]
```diff
@@ -6,6 +6,8 @@ import { HomeStack } from 'features/home/navigation/HomeNavigator'
 import { env } from 'libs/environment'
 import { flushAllPromises } from 'tests/utils'
 
+import { reactQueryProviderHOC } from '../../../tests/reactQueryProviderHOC'
```
Comment (antoinewg) [inline src/features/home/pages/Home.test.tsx:9]: pareil

**Comment by Giregm** [comment]
top !

**Pull request merged**